### PR TITLE
Drone blocker ignores dead ghost roles

### PR DESCRIPTION
**Pull request by Elijahrane**
## About the PR 
The blocker ignoring bodies is intentional, but only ones that might be cloned. We don't really care if they clean up a dead mouse or something. 
**Screenshots**

We're gonna prevent this
image

**Changelog**

:cl: Rane
- fix: Drones will no longer be blocked from acting near dead ghost roles.

### CODE REVIEW

File: Content.Server/Drone/DroneSystem.cs [hunk 1]
```diff
@@ -175,6 +176,11 @@ private bool NonDronesInRange(EntityUid uid, DroneComponent component)
             var xform = Comp<TransformComponent>(uid);
             foreach (var entity in _lookup.GetEntitiesInRange(xform.MapPosition, component.InteractionBlockRange))
             {
+                // Filter out dead ghost roles. Dead normal players are intended to block.
+                if ((TryComp<MobStateComponent>(entity, out var entityMobState) && HasComp<GhostTakeoverAvailableComponent>(entity) && entityMobState.IsDead()))
+                    continue;
```
Comment (mirrorcult) [inline Content.Server/Drone/DroneSystem.cs:181]: this should really be under the second if below this, since its a little slow

**Review by mirrorcult — CHANGES_REQUESTED** [review]

**Comment by Elijahrane** [comment]
Reviews addressed

File: Content.Server/Drone/DroneSystem.cs [hunk 2]
```diff
@@ -175,8 +176,12 @@ private bool NonDronesInRange(EntityUid uid, DroneComponent component)
             var xform = Comp<TransformComponent>(uid);
             foreach (var entity in _lookup.GetEntitiesInRange(xform.MapPosition, component.InteractionBlockRange))
             {
+                // Return true if the entity is/was controlled by a player and is not a drone or ghost.
                 if (HasComp<MindComponent>(entity) && !HasComp<DroneComponent>(entity) && !HasComp<GhostComponent>(entity))
```
Comment (HoofedEar) [inline Content.Server/Drone/DroneSystem.cs:180]:
```suggestion
                if (HasComp<MobStateComponent>(entity) && !HasComp<DroneComponent>(entity) && !HasComp<GhostComponent>(entity))
```
Not all player controlled things should block a drone's actions (such as vending machines). I tested this change on my local repo

Comment (Elijahrane) [reply Content.Server/Drone/DroneSystem.cs:180]: read your laws

**Review by mirrorcult — APPROVED** [review]

**Pull request merged**